### PR TITLE
Add test to build pipeline

### PR DESCRIPTION
**Pull request by SuperQ**
* Run all tests in Travis.
* Use `dep ensure` to install deps.
* Update deps.
* Misc Makefile cleanup.

Closes: https://github.com/Comcast/trickster/issues/75

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -41,16 +51,16 @@ docker-release:
...
-	go get github.com/alicebob/miniredis
-	go test -run '' -o ${GOPATH}/bin/trickster -v
+.PHONY: test
+test: deps
+	go test -o ${GOPATH}/bin/trickster -v ./...
 
-test-cover:
-	go get github.com/alicebob/miniredis
-	go test -run '' -o ${GOPATH}/bin/trickster -coverprofile=cover.out
+.PHONY: test-cover
```
Comment (juliusv) [inline Makefile]: Why define some `.PHONY`s above the target and others in a list at the bottom? Should be consistent IMO unless there's a reason.

Comment (SuperQ) [reply Makefile]: Yea, I was trying to move towards having it be with the targets.

Comment (juliusv) [reply Makefile]: Then IMO do it all at once...

Comment (SuperQ) [reply Makefile]: Done

**Comment by juliusv** [comment]
:+1:

**Pull request merged**